### PR TITLE
Fix and improve GCS2BQ blueprint

### PR DESCRIPTION
**Pull request by lcaggio**
In this PR I improve the GCS2BQ blueprint.
 - Fix roles
 - Implement Shared VPC roles relying on project `service_identity_iam` variable
 - Group project variable into `project_config` variable
 - Rely on optional variables to avoid `try()`

### CODE REVIEW

File: blueprints/data-solutions/gcs-to-bq-with-least-privileges/main.tf [hunk 1]
```diff
@@ -120,19 +79,14 @@ module "project" {
     "storage.googleapis.com",
     "storage-component.googleapis.com",
   ]
-
-  # additive IAM bindings avoid disrupting bindings in existing project
-  iam          = var.project_create != null ? local.iam : {}
-  iam_additive = var.project_create == null ? local.iam : {}
-  shared_vpc_service_config = local.shared_vpc_project == null ? null : {
+  group_iam    = local.group_iam
```
Comment (wiktorn) [inline blueprints/data-solutions/gcs-to-bq-with-least-privileges/main.tf:82]: `group_iam` is authoritative, so distinction below between `iam` and `iam_additive` will not suffice

File: blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf [hunk 2]
```diff
@@ -42,18 +43,18 @@ variable "prefix" {
...
-  default = null
-}
-
-variable "project_id" {
-  description = "Project id, references existing project if `project_create` is null."
-  type        = string
+  nullable = false
+  validation {
+    condition     = var.project_config.billing_account_id != null || var.project_config.project_id != null
+    error_message = "At least one of project_config.billing_account_id or var.project_config.project_ids should be set."
```
Comment (wiktorn) [inline blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf:56]: Typo in var.project_config.project_id**s**

File: blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf [hunk 3]
```diff
@@ -42,18 +43,18 @@ variable "prefix" {
   }
 }
 
-variable "project_create" {
-  description = "Provide values if project creation is needed, uses existing project if null. Parent is in 'folders/nnn' or 'organizations/nnn' format."
+variable "project_config" {
+  description = "Provide 'billing_account_id' value if project creation is needed, uses existing 'project_id' if null. Parent is in 'folders/nnn' or 'organizations/nnn' format."
```
Comment (wiktorn) [inline blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf:47]: Maybe worth mentioning, that if the project will be created, `var.prefix` will be used (contrary to using existing project)

**Review by wiktorn — APPROVED** [review]

File: blueprints/data-solutions/gcs-to-bq-with-least-privileges/main.tf (hunk 1 above)
Comment (lcaggio) [reply blueprints/data-solutions/gcs-to-bq-with-least-privileges/main.tf:82]: moved all to iam and iam_additive

File: blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf [hunk 4]
```diff
@@ -19,18 +19,19 @@ variable "cmek_encryption" {
 }
 
 variable "data_eng_principals" {
-  description = "Groups with Service Account Token creator role on service accounts in IAM format, eg 'group:group@domain.com'."
+  description = "Groups with Service Account Token creator role on service accounts, eg 'group@domain.com'."
```
Comment (wiktorn) [inline blueprints/data-solutions/gcs-to-bq-with-least-privileges/variables.tf:22]: I guess now this principals will have wider permissions, might be worth documenting here. Also, I think that they need to have a `group:` prefix now as they have moved back to `iam` / `iam_additive`

**Pull request merged**